features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthScreen.kt

```diff
@@ -68,8 +68,10 @@ fun AuthScreen(
                     failureTextId = R.string.auth_failure_label,
                     modifier = Modifier.fillMaxWidth()
                 ) { onEventSent(AuthContract.Event.Auth) }
-                Succeeded -> { /* the user is sent to onboarding/main screen */
-                    Text("Logged in but nothing to do...")
+                Succeeded -> {
+                    navController.navigate(AppDestinations.Home.landing) {
+                        popUpTo(AppDestinations.Features.home) { inclusive = true}
+                    }
```

Comment:
@lukstbit there is a issue here the events keeps getting emitted and so the nav keeps happening, one can see it by the flickering on the screen.

Do you have a suggestion here for a best approach so we don't leak memory?

Reply:
@alexandru-calinoiu That's happening because we are calling  navController.navigate() every time the composable is composed and it kind of enters a loop when the navigation runs its animations(and recomposes).
We need to use a _LaunchedEffect:_
```

LaunchedEffect(Unit) {
    controller.navigate(AppDestinations.Home.landing) {
        popUpTo(AppDestinations.Features.home) { inclusive = true }
     }
}
```

where _controller_ should be this statement at the top of _AuthScreen_:

`val controller by rememberUpdatedState(newValue = navController)`